### ISSUE TITLE
Use correct reply-to header in form page emails

### ISSUE THREAD

**Issue by balazs-endresz** [torchbox/wagtail#2198]
When a form is submitted then people sometimes reply to the notification email directly, and depending on the from_address field or settings.DEFAULT_FROM_EMAIL, others might get that reply too. Another variation of this is when they only copy in the intended recipient of the message but still send an email to the from_address too. Having a correct reply-to header in the notification email would solve this. However, that can't just be another field in the wagtailadmin, it has to be set to the email address of the person who submitted the form.

**Comment by kaedroho** [comment]
Thanks @balazs-endresz 

Could auto-reply emails become a problem here? It could be quite annoying/confusing if a few of the administrators went on holiday when someone submitted a form which would lead them to recieve some "I'm on holday" messages in response to their submission.

**Comment by balazs-endresz** [comment]
That's true, and sending automated replies to people who submit a form is probably even worse than the original issue.

So, I guess we shouldn't do this but I think it would also help if we could warn people in the notification emails about this, i.e. to remove the from address (`DEFAULT_FROM_EMAIL`) when replying to the user. Currently, I don't think the notification email text is customisable like that: https://github.com/torchbox/wagtail/blob/master/wagtail/wagtailforms/models.py#L226

**Comment by kevindice** [comment]
This feature would be useful.  Otherwise, I will have to write a plain view + form to accomplish the same thing.

**Comment by kevindice** [comment]
Maybe this could just be noted in the documentation.  I managed to do work out a "good enough" solution.  It makes an assumption that there is a field named "email" in the form, but if that assumption is wrong, it fails gracefully.

```python
class FormPage(MenuPage, AbstractEmailForm):
    intro = StreamField(StandardStreamBlock())
    thank_you_text = RichTextField(blank=True)

    content_panels = AbstractEmailForm.content_panels + [
        FormSubmissionsPanel(),
        StreamFieldPanel('intro'),
        InlinePanel('form_fields', label="Form fields"),
        FieldPanel('thank_you_text', classname="full"),
        MultiFieldPanel([
            FieldRowPanel([
                FieldPanel('from_address', classname="col6"),
                FieldPanel('to_address', classname="col6"),
            ]),
            FieldPanel('subject'),
        ], "Email"),
    ]

    def process_form_submission(self, form):
        if form.cleaned_data.get('email', False) and self.to_address:
            self.from_address = form.cleaned_data.get('email')
            self.send_mail(form)
        submission = super(FormPage, self).process_form_submission(form)                                                                     
        return submission

```

**Comment by lery12** [comment]
@kevindice 
This option would work if your SMTP provider allows sending mails from different e-mails.
Most reputable SMTP providers won't allow this.

The "reply_to" header has to get set in ..\wagtail\admin\mail.py function send_mail

**Comment by flurl** [comment]
I overwrote `EmailFormMixin.send_mail()` to pass a `reply_to` parameter to wagtail's `send_mail()`
```python
def send_mail(self, form):
  addresses = [x.strip() for x in self.to_address.split(",")]
  send_mail(
      self.subject,
      self.render_email(form),
      addresses,
      self.from_address,
      reply_to = (form.cleaned_data.get('email'), )
  )
```
